enhancements/authentication/AuthConfig-missing-fields.md

```diff
@@ -0,0 +1,500 @@
+      oidcProviders:
+      - name: foo
+        claimMappings:
+          uid:
+            expression: claims.email
+  ```
+-------------------------------------
+```yaml
+    apiVersion: config.openshift.io/v1
+    kind: Authentication
+    metadata:
+      name: cluster
+    spec:
+      type: OIDC
+      oidcProviders:
+      - name: foo
+        claimMappings:
+          uid:
+            claim: email
+```
+
+### 4. Adding Extra Claims for Role-Based Permissions
+**Scenario:**  
+A cluster administrator wants to utilize an external webhook authorizer to process custom claims (such as `role`) from the OIDC token for access control decisions.
+
+**Steps:**
+
+1. Update the authentication CRD to include extra claims:
+```yaml
+   apiVersion: config.openshift.io/v1
+   kind: Authentication
+   metadata:
+     name: cluster
+   spec:
+    type: OIDC
+    oidcProviders:
+      claimMappings:
+        extra:
+        - key: 'example.com/role'
+          valueExpression: 'claims.role'
+```
+
+#### 5. Enforcing Token Expiration Limits (Claim Validation)  
+**Scenario:**  
+A security-conscious organization wants to ensure that all OIDC tokens used for authentication have a maximum expiration time of 24 hours.  
+
+**Steps:**  
+1. Update the authentication CRD to enforce a maximum token expiration time:  
+   ```yaml
+   apiVersion: config.openshift.io/v1
+   kind: Authentication
+   metadata:
+     name: cluster
+   spec:
+    type: OIDC
+    oidcProviders:
+      claimValidationRules:
+     - expression: 'claims.exp - claims.nbf <= 86400'
+        message: total token lifetime must not exceed 24 hours
+  ```
+
+#### 6. Restricting Reserved Usernames (User Validation)  
+**Scenario:**  
+An OpenShift administrator wants to prevent users from being created with reserved system prefixes, such as `system:`, to avoid conflicts with system users.  
+
+**Steps:**  
+1. Update the authentication CRD to enforce username restrictions:  
+   ```yaml
+  apiVersion: config.openshift.io/v1
+  kind: Authentication
+  metadata:
+    name: cluster
+  spec:
+    type: OIDC
+    oidcProviders:
+      userValidationRules:
+      - expression: "!user.username.startsWith('system:')" # the expression will evaluate to true, so validation will succeed.
+          message: 'username cannot used reserved system: prefix'
+  ```
+
+### API Extensions
+To facilitate the configuration and validation of token claims, token issuers, and user validation rules, the existing `authentications.config.openshift.io` CRD is extended with new fields and structures. These extensions provide enhanced flexibility in token validation, token claim mappings, issuer configuration, and user validation. The proposed changes introduce new fields that allow administrators to define custom authentication behaviors.  
+The proposed API changes have been submitted in [openshift/api#2230](https://github.com/openshift/api/pull/2230). Below are examples of how users will configure these new fields:  
+
+
```

Comment:
On all these changes, it might be helpful for reviewers to include:
- What this field does and why users would care to set it. Write this information in the same way you would write user documentation for this field.
- What constraints will be in place for a user attempting to set this field.